Documentation/project-docs/testing.md

```diff
@@ -0,0 +1,74 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
+We usually run tests in parallel, unless we add an exception to the root of the project. Therefore we need to be careful to avoid possible side-effects when manipulating static members (e.g. properties).
```

Comment:
I haven't found anything regarding attribute parallelizable in xunit: https://xunit.github.io/docs/running-tests-in-parallel.html. I added the link for more info.